src/cmd/api-service/builder/api_service_builder.go

```diff
@@ -0,0 +1,148 @@
+package builder
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/onflow/api-service/m/v2/cmd/engine"
+	"github.com/onflow/api-service/m/v2/cmd/service"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow/protobuf/go/flow/access"
+)
+
+func NewFlowAPIServiceBuilder() *FlowAPIServiceBuilder {
+	return &FlowAPIServiceBuilder{
+		FlowServiceBuilder: service.NewFlowServiceBuilder("api-service"),
+	}
+}
+
+type FlowAPIServiceCmd struct {
+	service.FlowService
+	ServiceConfig service.ServiceConfig
+	RpcEngine     *engine.RPC
+}
+
+type FlowAPIServiceBuilder struct {
+	*service.FlowServiceBuilder
+	RpcConf                 engine.Config
+	ApiTimeout              time.Duration
+	ProtocolNodeAddresses   []string
+	ProtocolNodePublicKeys  []string
+	ExecutionNodeAddresses  []string
+	ExecutionNodePublicKeys []string
+	FlowDpsListenPort       string
+	FlowDpsHostUrl          string
+	FlowDpsMaxCacheSize     uint64
+	Api                     access.AccessAPIServer
+	RpcEngine               *engine.RPC
+}
+
+// Initialize Parse and print API Service command line arguments.
+func (fsb *FlowAPIServiceBuilder) Initialize() error {
+	flags := &fsb.ServiceConfig.Flags
+	flags.StringVarP(&fsb.RpcConf.ListenAddr, "rpc-addr", "r", ":9000", "the address the GRPC server listens on")
+	flags.DurationVar(&fsb.ApiTimeout, "flow-api-timeout", 3*time.Second, "tcp timeout of the Flow API gRPC socket")
+	flags.StringSliceVar(&fsb.ProtocolNodeAddresses, "protocol-node-addresses", []string{}, "the network addresses of the bootstrap access nodes e.g. access-001.mainnet.flow.org:9653,access-002.mainnet.flow.org:9653")
+	flags.StringSliceVar(&fsb.ProtocolNodePublicKeys, "protocol-node-public-keys", []string{}, "the networking public key of the bootstrap access nodes (in the same order as the bootstrap node addresses) e.g. \"d57a5e9c5.....\",\"44ded42d....\"")
+	flags.StringSliceVar(&fsb.ExecutionNodeAddresses, "execution-node-addresses", []string{}, "the network addresses of the bootstrap access nodes e.g. access-001.mainnet.flow.org:9653,access-002.mainnet.flow.org:9653")
+	flags.StringSliceVar(&fsb.ExecutionNodePublicKeys, "execution-node-public-keys", []string{}, "the networking public key of the bootstrap access nodes (in the same order as the bootstrap node addresses) e.g. \"d57a5e9c5.....\",\"44ded42d....\"")
+	flags.StringVarP(&fsb.FlowDpsHostUrl, "flow-dps-host-url", "d", "127.0.0.1:5005", "host address for dps")
+	flags.StringVarP(&fsb.FlowDpsListenPort, "flow-dps-listen-address", "a", "127.0.0.1:5006", "address to listen for flow dps access")
```

Comment:
Are you sure we should have defaults? Don't want that much traffic to AN-2

Reply:
I refactored following a similar convention set with protocol and execution node addresses. Thanks.